cassandra/src/main/java/io/micronaut/configuration/cassandra/health/CassandraHealthIndicator.java

```diff
@@ -0,0 +1,154 @@
+
+    private static final SimpleStatement VALIDATION_SELECT = selectFrom("system", "local")
+            .column(COL_RELEASE_VERSION)
+            .column(COL_CLUSTER_NAME)
+            .column(COL_CQL_VERSION)
+            .column(COL_DATA_CENTER)
+            .column(COL_NATIVE_PROTOCOL_VERSION)
+            .column(COL_PARTITIONER)
+            .column(COL_RACK)
+            .build()
+            .setConsistencyLevel(ConsistencyLevel.LOCAL_ONE);
+
+    private static final Logger LOG = LoggerFactory.getLogger(CassandraHealthIndicator.class);
+
+    private final CqlSession cqlSession;
+
+    /**
+     * Default constructor.
+     *
+     * @param cqlSession The The cassandra {@link CqlSession} to query for details
+     */
+    public CassandraHealthIndicator(final CqlSession cqlSession) {
+        this.cqlSession = cqlSession;
+    }
+
+    @Override
+    protected Map<String, Object> getHealthInformation() {
+        Map<String, Object> detail = new LinkedHashMap<>();
+        Map<UUID, Node> nodes = cqlSession.getMetadata().getNodes();
+        detail.put("session", cqlSession.isClosed() ? "CLOSED" : "OPEN");
+        Optional<CqlIdentifier> opKeyspace = cqlSession.getKeyspace();
+        if (opKeyspace.isPresent()) {
+            detail.put("keyspace", opKeyspace.get());
+        }
+        detail.put("nodes_count", nodes.keySet().size());
+
+        Map<UUID, Map<String, Object>> nodesMap = new HashMap<>();
+        Map<NodeState, Integer> nodeStateMap = new EnumMap<>(NodeState.class);
+        boolean up = false;
+        int i = 0;
+        for (Map.Entry<UUID, Node> entry : nodes.entrySet()) {
+            UUID uuid = entry.getKey();
+            Node node = entry.getValue();
+            nodeStateMap.merge(node.getState(), 1, (a, b) -> a + b);
+            if (node.getState() == NodeState.UP) {
+                up = true;
+            }
+            if (i++ < 10) {
+                Map<String, Object> nodeMap = new HashMap<>();
+                Optional<InetSocketAddress> opBroadcastAddress = node.getBroadcastAddress();
+                if (opBroadcastAddress.isPresent()) {
+                    nodeMap.put("broadcast_address", opBroadcastAddress.get().getAddress());
+                }
+                nodeMap.put("state", node.getState());
+                nodeMap.put("distance", node.getDistance());
+                nodeMap.put("open_connections", node.getOpenConnections());
+                nodeMap.put("cassandra_version", node.getCassandraVersion());
+                nodeMap.put("datacenter", node.getDatacenter());
+                nodeMap.put("rack", node.getRack());
+                nodeMap.put("uptime_ms", node.getUpSinceMillis());
+                nodeMap.put("is_reconnecting", node.isReconnecting());
+                nodesMap.put(uuid, nodeMap);
+            }
+        }
+        detail.put("nodes_state", nodeStateMap);
+        if (nodesMap.size() > 0) {
+            detail.put("nodes", nodesMap);
```

Comment:
Maybe change the key to `nodes (10 max.)` to make it clear that it will only ever show 10 nodes max.